pkg/extension/sumologicextension/extension.go

```diff
@@ -592,6 +618,135 @@ func (se *SumologicExtension) sendHeartbeatWithHTTPClient(ctx context.Context, h
 	return nil
 }
 
+func getHostIpAddress() (string, error) {
+	c, err := net.Dial("udp", "8.8.8.8:53")
+	if err != nil {
+		return "", err
+	}
+
+	defer c.Close()
+	a := c.LocalAddr().(*net.UDPAddr)
+	h, _, err := net.SplitHostPort(a.String())
+	if err != nil {
+		return "", err
+	}
+
+	return h, nil
+}
+
+func (se *SumologicExtension) updateMetadataWithHTTPClient(ctx context.Context, httpClient *http.Client) error {
+	u, err := url.Parse(se.BaseUrl() + metadataUrl)
+	if err != nil {
+		return fmt.Errorf("unable to parse metadata URL %w", err)
+	}
+
+	info, err := host.Info()
+	if err != nil {
+		return err
+	}
+
+	ip, err := getHostIpAddress()
+	if err != nil {
+		return err
+	}
+
+	td := se.conf.CollectorFields
+	if td == nil {
+		td = map[string]interface{}{}
+	}
+
+	var buff bytes.Buffer
+	if err = json.NewEncoder(&buff).Encode(api.OpenMetadataRequestPayload{
+		HostDetails: api.OpenMetadataHostDetails{
+			Name:        info.Hostname,
+			OsName:      info.OS,
+			OsVersion:   info.PlatformVersion,
+			Environment: se.conf.CollectorEnvironment,
+		},
+		CollectorDetails: api.OpenMetadataCollectorDetails{
+			RunningVersion: "1.0.0",
+		},
+		NetworkDetails: api.OpenMetadataNetworkDetails{
+			HostIpAddress: ip,
+		},
+		TagDetails: td,
+	}); err != nil {
+		return err
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, u.String(), &buff)
+	if err != nil {
+		return fmt.Errorf("unable to create HTTP request %w", err)
+	}
+
+	addJSONHeaders(req)
+
+	se.logger.Info("Updating collector metadata",
+		zap.String("URL", u.String()),
+		zap.String("body", buff.String()))
+
+	res, err := httpClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("unable to send HTTP request: %w", err)
+	}
+	defer res.Body.Close()
+
+	switch res.StatusCode {
+	default:
+		var buff bytes.Buffer
+		if _, err := io.Copy(&buff, res.Body); err != nil {
+			return fmt.Errorf(
+				"failed to copy collector metadata response body, status code: %d, err: %w",
+				res.StatusCode, err,
+			)
+		}
+
+		se.logger.Warn("Metadata API error response",
+			zap.Int("status", res.StatusCode),
+			zap.String("body", buff.String()))
+
+		return fmt.Errorf("collector metadata request failed: %w",
+			ErrorAPI{
+				status: res.StatusCode,
+				body:   buff.String(),
+			},
+		)
+
+	case http.StatusUnauthorized:
+		return errUnauthorizedMetadata
+
+	case http.StatusNoContent:
```

Comment:
What does it mean when the API returns a 204? When no changes happened? It's a bit peculiar, because we're using a POST here, which in principle shouldn't ever result in a 204.

Reply:
I was just sloppy, will remove 👍